src/sql/parts/registeredServer/common/objectExplorerService.ts

```diff
@@ -18,6 +18,7 @@ import * as TelemetryKeys from 'sql/common/telemetryKeys';
 import * as TelemetryUtils from 'sql/common/telemetryUtilities';
 import { ITelemetryService } from 'vs/platform/telemetry/common/telemetry';
 import { warn, error } from 'sql/base/common/log';
+import { ServerTreeView } from 'sql/parts/registeredServer/viewlet/serverTreeView';
```

Comment:
it shouldn't be a reference to UI in the service . is there any way we can prevent this? for instance can the view calls the service when focus changed to update the current connection?

Reply:
I don't know if that's possible, since we don't get events when OE loses focus. At some level we need to be able to get information about the current selection, whether it is focused, etc, and I haven't yet found a better way to do that than by making the Object Explorer service aware of a `ServerTreeView` object right now. Any thoughts?

Reply:
@anthonydresser Leila suggested you might have some input here. Any thoughts on whether there's a better design for this than just making the `ServerTreeView` an attribute of `ObjectExplorerService` like I've done here?

Reply:
I would say bypass the object explorer service, since this doesn't really belong here anyways and creates circular dependency issues.  Instead add a property in the connection management service that is updated by this controller https://github.com/Microsoft/sqlopsstudio/blob/vbump/slickgrid/src/sql/parts/registeredServer/viewlet/serverTreeController.ts#L24 when we new selection is made. I would also recommend making the conversion to a valid connection profile at this point to isolate code. This isn't a perfect solution but it avoids a lot of the circular dependencies and code you have.